### PR TITLE
feat(QC): data models for qcreport

### PR DESCRIPTION
**Pull request by kulgan**

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -3,3 +3,4 @@ jupyter==1.0.0
 jupyter-client==4.2.2
 jupyter-console==4.1.1
 jupyter-core==4.1.0
+pytest
```
Comment (anmaxvl) [inline dev-requirements.txt]: probably pin?

File: gdcdatamodel/models/qcreport.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
+from sqlalchemy import Boolean, Integer, Column, DateTime, String, Enum, ForeignKey, text
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import relationship, validates
+
+
+Base = declarative_base()
+SEVERITY = Enum("FAILED", "WARNING", "PASSED", name="error_severity")
```
Comment (anmaxvl) [inline gdcdatamodel/models/qcreport.py]: my initial idea was to map this to whatever `bio_submitter_qc` has and that's `WARNING` and `FATAL`, but people didn't like `FATAL` and I think we decided to go with `CRITICAL` instead.

File: gdcdatamodel/models/qcreport.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+TEST_RUN_STATUS = Enum("PENDING", "RUNNING", "FINISHED" "FAILED", name="test_run_status")
```
Comment (anmaxvl) [inline gdcdatamodel/models/qcreport.py]: missing comma

File: gdcdatamodel/models/qcreport.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            self.id, self.test_type, self.status
+        )
+
+    def to_json(self):
+        return {
+            'id': self.id,
+            'project_id': self.project_id,
+            'entity_id': self.entity_id,
+            'test_type': self.test_type,
+            'status': self.status,
```
Comment (anmaxvl) [inline gdcdatamodel/models/qcreport.py]: we might need `date_created` `last_updated` in `ISO` format or something?

File: gdcdatamodel/models/qcreport.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        }
+
+
+class ValidationResult(Base):
+
+    __tablename__ = 'qc_validation_results'
+
+    id = Column(Integer, primary_key=True)
+
+    node_id = Column(String, nullable=False)
```
Comment (anmaxvl) [inline gdcdatamodel/models/qcreport.py]: maybe add length as you did for the `TestRun` table columns

File: test/test_qc_models.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+
+def test_create_validation_result(g, test_run):
+
+    with g.session_scope():
+        vr = models.qcreport.ValidationResult()
+        vr.node_id = str(uuid.uuid4())
+        vr.severity = "fatal"
+        vr.error_type = "NO_READ_PAIR_NUMBER"
+        vr.message = "he FASTQ is paired but has no read_pair_number"
```
Comment (anmaxvl) [inline test/test_qc_models.py]: `The`?

File: gdcdatamodel/models/qcreport.py [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+    __tablename__ = 'qc_validation_results'
+
+    id = Column(Integer, primary_key=True)
+
+    node_id = Column(String(64), nullable=False)
+
+    error_type = Column(String(128), nullable=True, default='')
+
+    node_type = Column(String(128), nullable=False)
```
Comment (anmaxvl) [inline gdcdatamodel/models/qcreport.py]: Probably put a comment as to what's expected here, i.e. `Node.label` or `Node._dictionary['Title']` or `Node.name`?

File: gdcdatamodel/gdc_postgres_admin.py [hunk 8]
```diff
@@ -420,10 +456,60 @@ def add_subcommand_revoke(subparsers):
 
 def get_parser():
     parser = argparse.ArgumentParser()
+    parser.add_argument(
```
Comment (anmaxvl) [inline gdcdatamodel/gdc_postgres_admin.py]: 👏

File: gdcdatamodel/models/qcreport.py (hunk 7 above)
Comment (anmaxvl) [reply gdcdatamodel/models/qcreport.py]: I think best is `Node.label`, since we can then directly map to endpoints.

**Comment by huarngpa** [comment]
👍

**Pull request merged**